### PR TITLE
feat(flagd): use test-harness version number for integration tests

### PR DESCRIPTION
**Pull request by aepfli**

### CODE REVIEW

File: providers/openfeature-provider-flagd/tests/e2e/flagd_container.py [hunk 1]
```diff
@@ -11,11 +12,15 @@
...
-        image: str = "ghcr.io/open-feature/flagd-testbed:v0.5.15",
+        image: str = "ghcr.io/open-feature/flagd-testbed",
         port: int = 8013,
         **kwargs,
     ) -> None:
-        super().__init__(image, **kwargs)
+        my_path = os.path.abspath(os.path.dirname(__file__))
+        path = os.path.join(my_path, "../../openfeature/test-harness/version.txt")
+        with open(path) as file:
+            data = file.read().rstrip()
```
Comment (gruebel) [inline providers/openfeature-provider-flagd/tests/e2e/flagd_container.py:22]: I recommend to use `Path` from `pathlib` here, it makes file handling much easier. There are couple of different ways to get to the path you want, so feel free to check it out
ex.
```suggestion
        path = Path(__file__).parents[2] / "openfeature/test-harness/version.txt"
        data = path.read_tex().rstrip()
```
you can also keep `../../` but then I recommend to invoke `.resolve()` once.

**Review by gruebel — COMMENTED** [review]
are the failing tests related to the upgrade to version `0.5.18`?

File: providers/openfeature-provider-flagd/tests/e2e/flagd_container.py (hunk 1 above)
Comment (aepfli) [reply providers/openfeature-provider-flagd/tests/e2e/flagd_container.py]: uh nice, that is a way cleaner api - thank you

**Comment by aepfli** [comment]
> are the failing tests related to the upgrade to version `0.5.18`?

It seems like there is something flaky here. I'll check, but I am already working on a different approach for tests with an unstable connection. We don't need an image that is regularly disconnecting, as we're using test containers; we can just kill the container and be faster with our tests. Additionally this should grant us more stability in the tests, as we're in control

**Review by gruebel — APPROVED** [review]

**Pull request merged**